### PR TITLE
[android][camera] Fix onBarCodeScanned `bounds` for ExpoCameraView

### PR DESCRIPTION
**Pull request by devon94**
# Why

The type definitions for BarcodeScanningResult contains `bounds`:
[expo/packages/expo-camera/src/CameraView.tsx](https://github.com/expo/expo/blob/main/packages/expo-camera/src/CameraView.tsx)

However, on Android the bounds from the CameraView (latest & legacy) is missing, it seems to be incorrectly sending `boundingBox` instead which leads to crashes
https://github.com/expo/expo/blob/b05ee6182f0a602a842aa720d76a74663de1147e/packages/expo-camera/android/src/main/java/expo/modules/camera/ExpoCameraView.kt#L556
https://github.com/expo/expo/blob/b275013e842d4f61b7d7594411a87fac4027373a/packages/expo-camera/android/src/main/java/expo/modules/camera/legacy/ExpoCameraView.kt#L288

On iOS, it seems like it is added properly in BarcodeScannerUtils
https://github.com/expo/expo/blob/b05ee6182f0a602a842aa720d76a74663de1147e/packages/expo-camera/ios/Current/BarcodeScannerUtils.swift

Minimal Reproduction repo (Latest Camera):

https://github.com/devon94/expo-camera-barcode-scanned-event-repro

# How

Fixed incorrect `boundingBox` being used for `BarcodeScannedEvent` by renaming to `bounds`

# Test Plan

Ran `yarn android` in `apps/bare-expo`

Tested fixed camera module against reproduction app to ensure it doesn't crash anymore
https://github.com/devon94/expo-camera-barcode-scanned-event-repro

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by LeiSurrre** [comment]
🙏🏻

File: packages/expo-barcode-scanner/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,8 @@
 
 - Add missing `react` and `react-native` peer dependencies for isolated modules. ([#30457](https://github.com/expo/expo/pull/30457) by [@byCedric](https://github.com/byCedric))
 
+- Fixed `bounds` incorrectly being returned as `boundingBox` in BarcodeScanningResult when using ExpoCameraView on Android. ([#30510](https://github.com/expo/expo/pull/30510) by [@devon94](https://github.com/devon94))
+
```
Comment (lukmccall) [inline packages/expo-barcode-scanner/CHANGELOG.md:14]:
```suggestion
- Fixed `bounds` incorrectly being returned as `boundingBox` in BarcodeScanningResult when using ExpoCameraView on Android. ([#30510](https://github.com/expo/expo/pull/30510) by [@devon94](https://github.com/devon94))

```

**Review by lukmccall — APPROVED** [review]
Looks good! Thanks for your contribution.

**Review by lukmccall — CHANGES_REQUESTED** [review]
The code doesn't compile. Please check the CI jobs.

**Comment by devon94** [comment]
That was my bad! Missed a part of the code in the initial PR 😅 That should be pushed and compiling now

**Comment by devon94** [comment]
Oh strange that it failed again, I think I probably missed something else. Just pushed. Sorry about that. Using a github client i'm not quite used to. Is there a way I can run the test suite on a a personal repo?

**Comment by devon94** [comment]
Looks like it passes now. Hopefully I ran a similar action to the one here

**Review by lukmccall — APPROVED** [review]
Nice! Thanks for your contribution!

**Pull request merged**